### PR TITLE
updated to support node manifest API v2

### PR DESCRIPTION
**Pull request by kathmbeck**
Removes logic that creates manifests depending on the `updatedAt` time within the source plugin and abstracts away this logic to the new node manifest API. 

Note that `updatedAt` is optional. If `updatedAt` is not included, we will create a manifest for every node that gets called with createNodeManifest.We default to creating node manifests for content updated in the last 30 days. To change that, set a NODE_MANIFEST_MAX_DAYS_OLD environment variable.

See documentation here for further details: https://www.gatsbyjs.com/docs/reference/config-files/actions/#unstable_createNodeManifest

### CODE REVIEW

**Comment by matjack1** [comment]
thank you @kathmbeck ! Maybe makes sense mentioning the `NODE_MANIFEST_MAX_DAYS_OLD` also in the documentation here? What do you think?

**Comment by kathmbeck** [comment]
> thank you @kathmbeck ! Maybe makes sense mentioning the `NODE_MANIFEST_MAX_DAYS_OLD` also in the documentation here? What do you think?

https://github.com/gatsbyjs/gatsby/pull/34166

**Review by DanielSLew — APPROVED** [review]
Looks good to me! Just that one change that might be needed that i dm'd you about

File: src/utils.js [hunk 1]
```diff
@@ -57,44 +59,45 @@ async function getLoader({ cache, loadStateFromCache, ...options }) {
...
+      }
       // Example manifestId: "34324203-2021-07-08T21:52:28.791+01:00"
-      const manifestId = `${node.entityPayload.id}-${node.entityPayload.meta.updated_at}`;
-
+      const manifestId = `${node.entityPayload.id}-${updatedAt}`;
+     
       unstable_createNodeManifest({
         manifestId,
         node,
+        updatedAt
```
Comment (DanielSLew) [inline src/utils.js:94]:
```suggestion
        updatedAt
```
```suggestion
        updatedAtUTC: updatedAt
```

**Comment by matjack1** [comment]
@kathmbeck just a heads up, not sure if you can do anything, but I had a project with Gatsby v4.0.0 and using this PR I got this error:

```
Missing onError handler for invocation 'extracting-queries', error was 'Error: ENOENT: no such file or directory, open
'...\gatsby-blog-demo\public\__node-manifests\gatsby-source-datocms\69028654-2021-12-07T11:13:23.915+00:0
0.json''. Stacktrace was 'Error: ENOENT: no such file or directory, open '...\gatsby-blog-demo\public\__n
ode-manifests\gatsby-source-datocms\69028654-2021-12-07T11:13:23.915+00:00.json''
```

And the build failed. Is this expected? Probably something changed on the file name format?

Upgrading to the latest Gatsby v4.3.0 fixes the error. Let me know if that's fine with you and we can merge this!

**Comment by kathmbeck** [comment]
Commenting here for reference. Per our discussion, this is a known issue with the Windows environment that was corrected with the upgrade.

**Pull request merged**